contracts/protocol/libraries/logic/AuctionLiquidationLogic.sol

```diff
@@ -0,0 +1,181 @@
+                    oracle: params.priceOracle
+                })
+            );
+
+        // check that users health is below threshold
+        require(
+            vars.healthFactor < THRESHOLD,
+            "Users health factor must be below the threshold"
+        );
+
+        DataTypes.Auction storage _auction = auctionData[
+            params.collateralAsset
+        ][params.collateralTokenId];
+
+        // set values
+        _auction.endTime = block.timestamp + 2 days;
+    }
+
+    /// @notice let anyone end an auction if the time is up
+    function executeAuctionLiquidationEnd(
+        mapping(address => mapping(uint256 => DataTypes.Auction))
+            storage auctionData,
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        mapping(uint256 => address) storage reservesList,
+        mapping(address => DataTypes.UserConfigurationMap) storage usersConfig,
+        DataTypes.ExecuteAuctionLiquidationEndParams memory params
+    ) external {
+        AuctionLiquidationEndLocalVars memory vars;
+
+        DataTypes.Auction storage _auction = auctionData[
+            params.collateralAsset
+        ][params.collateralTokenId];
+
+        // check if the auction is still going - check if time is up
+        require(
+            block.timestamp > _auction.endTime,
+            "Auction end time not reached"
+        );
+
+        // check if is still pending
+        require(!_auction.isResolved, "Auction is already resolved");
+
+        // set resolved
+        _auction.isResolved = true;
+
+        // check caller is highest bid
+        require(
+            params.bidder == _auction.highestBidder,
+            "Caller is not the highest bidder"
+        );
+
+        // check highest bid is above 95% of floor
+        vars.collateralPrice = IPriceOracleGetter(params.priceOracle)
+            .getAssetPrice(params.collateralAsset);
+        require(
+            _auction.highestBid > (95 * vars.collateralPrice) / 100,
+            "Highest bid is not above 95% of floor"
+        );
+
+        // check that collateral owner is still underwater
+        DataTypes.UserConfigurationMap storage userConfig = usersConfig[
+            params.user
+        ];
+
+        (, , , , , , , vars.healthFactor, , ) = GenericLogic
+            .calculateUserAccountData(
+                reservesData,
+                reservesList,
+                DataTypes.CalculateUserAccountDataParams({
+                    userConfig: userConfig,
+                    reservesCount: params.reservesCount,
+                    user: params.user,
+                    oracle: params.priceOracle
+                })
+            );
+
+        require(
+            vars.healthFactor < THRESHOLD,
+            "Users health factor must be below the threshold"
+        );
+
+        /// TODO: settle debt from liquidation
+
+        // repay debt
+        // or supply -> wrapped eth gw
+        // also pay fee
```

Comment:
@WalidOfNow can you help point me to an example of how we might supply this back to the market?

Reply:
https://github.com/para-space/paraspace-core/blob/777a606a6a68154fe3882a83226d812b0d3b8d48/contracts/protocol/libraries/logic/LiquidationLogic.sol#L387